src/io/setup.rs

```diff
@@ -89,14 +89,29 @@ impl IoSetup {
     }
 }
 
+
+/// Where does the "primary input" stream come from?
+enum PrimaryInputMode {
```

Comment:
This seems to duplicate code from `driver.rs`. Wouldn't `PrimaryInputMode::Undefined` be better expressed as `None` for `Option<PrimaryInputMode>`?

Reply:
Yes, it duplicates a bit. But in both cases, the types are private to their respective modules — they are more or less implementation details of the "builder"-type objects. Because they barely come with any code, I felt it was better to duplicate a bit than to try share the type within the crate. Reasonable enough?

Reply:
That makes sense. I did not see the big picture.